Tests/IntegrationTests/Card Tests/Card Scanner/CardScannerControllerTests.swift

```diff
@@ -98,19 +116,22 @@
 
         // MARK: - Helpers
 
-        private func makeSUT() -> (CardScannerController, UIViewController, CardScannerProviderSpy) {
+        private func makeSUT(isAvailable: Bool = true) -> (CardScannerController, UIViewController, CardScannerProviderSpy) {
             let presenter = UIViewController()
             let cardScanner = CardScannerProviderSpy()
 
             let sut = CardScannerController(
                 presenter: presenter,
-                availabilityProvider: CardScannerAvailalabilityMock(),
-                cardScannerProvider: cardScanner, analyticsHandler: self.sendEvent
+                availabilityProvider: CardScannerAvailalabilityMock(isScannerAvailable: isAvailable),
+                cardScannerProvider: cardScanner,
+                analyticsHandler: sendEvent
             )
             return (sut, presenter, cardScanner)
         }
 
-        private func sendEvent(_ subtype: AnalyticsEventLog.LogSubType) {}
+        private func sendEvent(_ subtype: AnalyticsEventLog.LogSubType) {
+            currentEventSubtype = subtype
```

Comment:
In the case we could test events array and their order, here we could just append events to the array intead